tests/js/spec/components/modals/inviteMembersModal.spec.jsx

```diff
@@ -0,0 +1,256 @@
+import {Modal} from 'react-bootstrap';
+import React from 'react';
+
+import {mount} from 'enzyme';
+import InviteMembersModal from 'app/components/modals/inviteMembersModal';
+
+describe('InviteMembersModal', function() {
+  const org = TestStubs.Organization();
+
+  const roles = [
+    {
+      id: 'admin',
+      name: 'Admin',
+      desc: 'This is the admin role',
+    },
+    {
+      id: 'member',
+      name: 'Member',
+      desc: 'This is the member role',
+    },
+  ];
+
+  MockApiClient.addMockResponse({
+    url: `/organizations/${org.slug}/members/me/`,
+    method: 'GET',
+    body: {roles},
+  });
+
+  it('renders', async function() {
+    const wrapper = mount(
+      <InviteMembersModal
+        Body={Modal.Body}
+        Header={Modal.Header}
+        Footer={Modal.Footer}
+        organization={org}
+      />,
+      TestStubs.routerContext()
+    );
+
+    // Starts with one invite row
+    expect(wrapper.find('StyledInviteRow')).toHaveLength(1);
+
+    // We have two roles loaded from the members/me endpoint, defaulting to the
+    // 'member' role.
+    expect(wrapper.find('RoleSelectControl').props().options).toHaveLength(roles.length);
+    expect(wrapper.find('RoleSelectControl Value').text()).toBe('Member');
+  });
+
+  it('can add a second row', async function() {
+    const wrapper = mount(
+      <InviteMembersModal
+        Body={Modal.Body}
+        Header={Modal.Header}
+        Footer={Modal.Footer}
+        organization={org}
+      />,
+      TestStubs.routerContext()
+    );
+
+    expect(wrapper.find('StyledInviteRow')).toHaveLength(1);
+    wrapper.find('AddButton').simulate('click');
+    expect(wrapper.find('StyledInviteRow')).toHaveLength(2);
+  });
+
+  it('errors on duplicate emails', async function() {
+    const wrapper = mount(
+      <InviteMembersModal
+        Body={Modal.Body}
+        Header={Modal.Header}
+        Footer={Modal.Footer}
+        organization={org}
+      />,
+      TestStubs.routerContext()
+    );
+
+    wrapper.find('AddButton').simulate('click');
+    expect(wrapper.find('StyledInviteRow')).toHaveLength(2);
+
+    const rows = wrapper.find('StyledInviteRow');
+
+    rows
```

Comment:
Can we change the Select controls here instead of just calling the callbacks? `tests/js/helpers/select` might be helpful?

Reply:
I'll give the helper a try. I was having a bit of trouble getting it to propagate.